src/main/java/com/o19s/es/ltr/feature/store/PrecompiledTemplateFeature.java

```diff
@@ -80,36 +81,48 @@ public String name() {
     }
 
     @Override
-    public Query doToQuery(QueryShardContext context, FeatureSet set, Map<String, Object> params) {
+    public Query doToQuery(LtrQueryContext context, FeatureSet set, Map<String, Object> params) {
         List<String> missingParams = queryParams.stream()
                 .filter((x) -> params == null || !params.containsKey(x))
                 .collect(Collectors.toList());
-        if (!missingParams.isEmpty()) {
+        if (context.isFeatureActive(name) && !missingParams.isEmpty()) {
```

Comment:
instead of checking this on every Feature implementation I'd do this check on all the FeatureSet implementations in the toQueries method:
```java
        List<Query> queries = new ArrayList<>(features.size());
        for(Feature feature : features) {
            if (context.isFeatureActive(name)) {
               queries.add(feature.doToQuery(context, this, params));
            } else {
               queries.add(new MatchNoDocsQuery( "Feature " + feature.name() + " deactivated" ));
            }
        }
        return queries;
```

The reason is that it's likely we'll add more Feature implementation and it'll be easy to forget to do this check in the future.
Adding new FeatureSet implementations seems unlikely.

Reply:
Makes sense. done